server/preferences.go

```diff
@@ -28,19 +32,39 @@ func (p *Plugin) setPrimaryPlatform(userID string, primaryPlatform string) error
 		return fmt.Errorf("invalid primary platform: %s", primaryPlatform)
 	}
 
-	appErr := p.API.UpdatePreferencesForUser(userID, []model.Preference{{
-		UserId:   userID,
-		Category: PreferenceCategoryPlugin,
-		Name:     storemodels.PreferenceNamePlatform,
-		Value:    primaryPlatform,
-	}})
+	appErr := p.updatePreferenceForUser(userID, storemodels.PreferenceNamePlatform, primaryPlatform)
 	if appErr != nil {
 		return appErr
 	}
+	return nil
+}
+
+func (p *Plugin) getNotificationPreference(userID string) bool {
+	// this call returns a generic error if the preference does not exist,
+	// we can omit the error check here and return off by default
+	pref, _ := p.API.GetPreferenceForUser(userID, PreferenceCategoryPlugin, storemodels.PreferenceNameNotification)
+	return pref.Value == storemodels.PreferenceValueNotificationOn
+}
+
+func (p *Plugin) setNotificationPreference(userID string, enable bool) error {
+	value := storemodels.PreferenceValueNotificationOff
+	if enable {
+		value = storemodels.PreferenceValueNotificationOn
+	}
+
+	err := p.updatePreferenceForUser(userID, storemodels.PreferenceNameNotification, value)
```

Comment:
changed to appErr. 

And I agree, `Preferences` are missimg from the `pluginapi`! I actually wondered why we used this rather than the KV store, but I suppose we get access to preferences from redux so it's easier for frontend components.